docs/contrib/PlatformBringUp.md

```diff
@@ -0,0 +1,415 @@
+
+
+```
+DESIGN_CONFIG=./designs/MyNewPlatform/riscv32i/config.mk
+```
+
+
+The `config.mk` file will be generated later in the [Design Directory](#DesignDirectory) section of this document.
+
+
+## Platform Directory
+
+Create a directory for the new technology inside `OpenROAD-flow-scripts/flow/platforms` to contain the necessary files for the OpenROAD flow.
+
+
+```
+$ mkdir OpenROAD-flow-scripts/flow/platforms/MyNewPlatform
+```
+
+
+
+## Design Directory
+
+The design directory contains the configuration files for all the designs of a specific platform. Create a directory for the new platform in OpenROAD-flow-scripts/flow/designs to contain the relevant files and directories for all the designs for the flow in that specific platform. Each design requires its own `config.mk` and `constraints.sdc` files. 
+
+Follow the steps below to create the necessary directories and files. **NOTE: riscv32i is just an example and not a required name.**:
+
+
+```
+$ mkdir -p OpenROAD-flow-scripts/flow/designs/MyNewPlatform/riscv32i
+$ touch OpenROAD-flow-scripts/flow/designs/MyNewPlatform/riscv32i/config.mk
+$ touch OpenROAD-flow-scripts/flow/designs/MyNewPlatform/riscv32i/constraints.sdc
+```
+
+
+This creates two directories MyNewPlatform and riscv32i and two empty files `config.mk` and `constraints.sdc` in OpenROAD-flow-scripts/flow/designs/MyNewPlatform/riscv32i. 
+
+
+## Platform Configuration
+
+This section describes the necessary files in the platform directory needed for the OpenROAD flow. Specifically the `config.mk` file in the platform directory has all of the configuration variables that the flow uses. Refer to the OpenROAD-flow-scripts documentation for a full list of configuration variables that can be set.
+
+For an example of a platform config.mk file, refer to `OpenROAD-flow-scripts/flow/platforms/sky130hd/config.mk.`
+
+
+## Design Configuration
+
+This section describes files in the design directory.
+
+
+## config.mk
+
+The `config.mk` file describes design-specific variables.
+
+For Example:
+
+
+```
+    DESIGN_NAME
+    PLATFORM
+    VERILOG_FILES
+    SDC_FILE
+    CORE_UTILIZATION
+    CORE_ASPECT_RATIO
+    CORE_MARGIN
+    PLACE_DENSITY
+```
+
+Alternatively, in place of `CORE_UTILIZATION` and `CORE_ASPECT_RATIO`, `DIE_AREA` and `CORE_AREA` can be specified instead.
+
+
+Following is a sample `config.mk` file for the `riscv32i` design:
+
+
+```
+#config.mk
+###########################
+export DESIGN_NAME     = riscv
+export PLATFORM        = sky130hd
+
+export VERILOG_FILES = $(sort $(wildcard ./designs/src/$(DESIGN_NICKNAME)/*.v))
+
+export SDC_FILE = ./designs/$(PLATFORM)/$(DESIGN_NICKNAME)/constraint.sdc
```

Comment:
`DESIGN_NICKNAME` hasn't been defined. It needs to be `DESIGN_NAME` here.

Reply:
This is quite confusing but this is how it is currently setup in OpenROAD-flow-scripts and actually does work. If you check line 169 in the Makefile - `DESIGN_NICKNAME` is set to `DESIGN_NAME`. I've noticed that there is no consistency between platforms whether it uses `DESIGN_NAME` or `DESIGN_NICKNAME` when setting the paths. sky130hd uses `DESIGN_NICKNAME` which is where this example was pulled from

Reply:
Right, this works because of deferred variable expansion. I don't really feel like it's intended to be that way though. The whole purpose of `DESIGN_NAME` vs. `DESIGN_NICKNAME` is that `DESIGN_NAME` needs to correspond to the top module name. This name can often be non-descriptive due to the codebase (`soc_top`, `bsg_chip`, etc.) or even conflict with different designs with the same top module name. That's why we have `DESIGN_NICKNAME` to disambiguate and clarify the names, because `$(DESIGN_NICKNAME)` is the name used for generated directories, etc.

tl;dr: I think that `DESIGN_NICKNAME` should only be used if it exists, and it should only exist if it's different from `DESIGN_NAME`. Perhaps that means there's some flow cleaning to do, but that needs to be done anyways.

Reply:
Ah okay that makes sense. I went ahead and changed the example to use `DESIGN_NAME` instead of `DESIGN_NICKNAME`.